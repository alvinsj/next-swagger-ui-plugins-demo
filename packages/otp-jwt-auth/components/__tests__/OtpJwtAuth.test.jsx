import React from 'react'
import OtpJwtAuth from '../OtpJwtAuth'
import renderer from 'react-test-renderer'
import { Map } from 'immutable'

describe('<OtpJwtAuth />', () => {
  const otpJwtAuthSelectors = { 
    email: () => 'test@example.com', 
    otp: () => 'example-otp',
    isOtpSent: () => false
  }
  const authSelectors = {
    authorized: () => ({ get: () => false })
  }
  const errSelectors = {
    allErrors: () => new Map()
  }

  const getComponent = (name) => 
    function GetComponent({children, ...props}) { 
      return <mock-element className={name} data-props={props}>{children}</mock-element> 
    }

  const system = {
    otpJwtAuthSelectors, 
    authSelectors,
    errSelectors,
    getComponent
  }

  it('renders form', () => {
    
    const tree = renderer
    .create(
      <OtpJwtAuth 
        {...system}
        getSystem={() => system}
        schema={{}}
        name="OtpJwtAuth"
      />)
    .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders otp sent message', () => {
    otpJwtAuthSelectors.isOtpSent = () => true
    const tree = renderer
    .create(
      <OtpJwtAuth 
        {...system}
        getSystem={() => system}
        schema={{}}
        name="OtpJwtAuth"
      />)
    .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders error messages', () => {
    otpJwtAuthSelectors.isOtpSent = () => false
    errSelectors.allErrors = () => new Map({'error': {get: () => 'OtpJwtAuth'}})

    const tree = renderer
      .create(
        <OtpJwtAuth 
          {...system}
          getSystem={() => system}
          schema={{}}
          name="OtpJwtAuth"
        />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders logout', () => {
    otpJwtAuthSelectors.isOtpSent = () => false
    errSelectors.allErrors = () => new Map()
    authSelectors.authorized = () => ({ get: () => true })

    const tree = renderer
      .create(
        <OtpJwtAuth 
          {...system}
          getSystem={() => system}
          schema={{}}
          name="OtpJwtAuth"
        />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
