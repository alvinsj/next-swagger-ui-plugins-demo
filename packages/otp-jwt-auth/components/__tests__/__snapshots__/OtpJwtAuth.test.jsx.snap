// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OtpJwtAuth /> renders error messages 1`] = `
[
  <mock-element
    className="Row"
    data-props={{}}
  >
    <strong>
      OTP Login
    </strong>
  </mock-element>,
  <mock-element
    className="Row"
    data-props={{}}
  >
    <label
      htmlFor="jwt_email"
    >
      Email:
    </label>
    <mock-element
      className="Col"
      data-props={{}}
    >
      <mock-element
        className="Input"
        data-props={
          {
            "data-name": "email",
            "disabled": false,
            "id": "jwt_email",
            "onChange": [Function],
            "required": "required",
            "type": "email",
            "value": "test@example.com",
          }
        }
      />
    </mock-element>
  </mock-element>,
  <mock-element
    className="Row"
    data-props={{}}
  >
    <label
      htmlFor="jwt_otp"
    >
      OTP:
    </label>
    <mock-element
      className="Col"
      data-props={{}}
    >
      <mock-element
        className="Input"
        data-props={
          {
            "data-name": "otp",
            "disabled": false,
            "id": "jwt_otp",
            "onChange": [Function],
            "required": "required",
            "type": "text",
            "value": "example-otp",
          }
        }
      />
    </mock-element>
  </mock-element>,
  <div
    className="auth-msg-wrapper"
  >
    <mock-element
      className="authError"
      data-props={
        {
          "error": {
            "get": [Function],
          },
        }
      }
    />
  </div>,
  <div
    className="auth-btn-wrapper"
  >
    <div>
      <mock-element
        className="Button"
        data-props={
          {
            "className": "btn modal-btn auth send-otp",
            "disabled": false,
            "onClick": [Function],
          }
        }
      >
        Send OTP
      </mock-element>
      <mock-element
        className="Button"
        data-props={
          {
            "className": "btn modal-btn auth authorize",
            "disabled": false,
            "onClick": [Function],
          }
        }
      >
        Authorize
      </mock-element>
    </div>
  </div>,
]
`;

exports[`<OtpJwtAuth /> renders form 1`] = `
[
  <mock-element
    className="Row"
    data-props={{}}
  >
    <strong>
      OTP Login
    </strong>
  </mock-element>,
  <mock-element
    className="Row"
    data-props={{}}
  >
    <label
      htmlFor="jwt_email"
    >
      Email:
    </label>
    <mock-element
      className="Col"
      data-props={{}}
    >
      <mock-element
        className="Input"
        data-props={
          {
            "data-name": "email",
            "disabled": false,
            "id": "jwt_email",
            "onChange": [Function],
            "required": "required",
            "type": "email",
            "value": "test@example.com",
          }
        }
      />
    </mock-element>
  </mock-element>,
  <mock-element
    className="Row"
    data-props={{}}
  >
    <label
      htmlFor="jwt_otp"
    >
      OTP:
    </label>
    <mock-element
      className="Col"
      data-props={{}}
    >
      <mock-element
        className="Input"
        data-props={
          {
            "data-name": "otp",
            "disabled": false,
            "id": "jwt_otp",
            "onChange": [Function],
            "required": "required",
            "type": "text",
            "value": "example-otp",
          }
        }
      />
    </mock-element>
  </mock-element>,
  <div
    className="auth-btn-wrapper"
  >
    <div>
      <mock-element
        className="Button"
        data-props={
          {
            "className": "btn modal-btn auth send-otp",
            "disabled": false,
            "onClick": [Function],
          }
        }
      >
        Send OTP
      </mock-element>
      <mock-element
        className="Button"
        data-props={
          {
            "className": "btn modal-btn auth authorize",
            "disabled": false,
            "onClick": [Function],
          }
        }
      >
        Authorize
      </mock-element>
    </div>
  </div>,
]
`;

exports[`<OtpJwtAuth /> renders logout 1`] = `
[
  <mock-element
    className="Row"
    data-props={{}}
  >
    <strong>
      OTP Login
    </strong>
  </mock-element>,
  <mock-element
    className="Row"
    data-props={{}}
  >
    <label
      htmlFor="jwt_email"
    >
      Email:
    </label>
    <mock-element
      className="Col"
      data-props={{}}
    >
      <mock-element
        className="Input"
        data-props={
          {
            "data-name": "email",
            "disabled": true,
            "id": "jwt_email",
            "onChange": [Function],
            "required": "required",
            "type": "email",
            "value": "test@example.com",
          }
        }
      />
    </mock-element>
  </mock-element>,
  <mock-element
    className="Row"
    data-props={{}}
  >
    <label
      htmlFor="jwt_otp"
    >
      OTP:
    </label>
    <mock-element
      className="Col"
      data-props={{}}
    >
      <mock-element
        className="Input"
        data-props={
          {
            "data-name": "otp",
            "disabled": true,
            "id": "jwt_otp",
            "onChange": [Function],
            "required": "required",
            "type": "text",
            "value": "example-otp",
          }
        }
      />
    </mock-element>
  </mock-element>,
  <div
    className="auth-btn-wrapper"
  >
    <mock-element
      className="Button"
      data-props={
        {
          "className": "btn modal-btn auth authorize",
          "onClick": [Function],
        }
      }
    >
      Logout
    </mock-element>
  </div>,
]
`;

exports[`<OtpJwtAuth /> renders otp sent message 1`] = `
[
  <mock-element
    className="Row"
    data-props={{}}
  >
    <strong>
      OTP Login
    </strong>
  </mock-element>,
  <mock-element
    className="Row"
    data-props={{}}
  >
    <label
      htmlFor="jwt_email"
    >
      Email:
    </label>
    <mock-element
      className="Col"
      data-props={{}}
    >
      <mock-element
        className="Input"
        data-props={
          {
            "data-name": "email",
            "disabled": false,
            "id": "jwt_email",
            "onChange": [Function],
            "required": "required",
            "type": "email",
            "value": "test@example.com",
          }
        }
      />
    </mock-element>
  </mock-element>,
  <mock-element
    className="Row"
    data-props={{}}
  >
    <label
      htmlFor="jwt_otp"
    >
      OTP:
    </label>
    <mock-element
      className="Col"
      data-props={{}}
    >
      <mock-element
        className="Input"
        data-props={
          {
            "data-name": "otp",
            "disabled": false,
            "id": "jwt_otp",
            "onChange": [Function],
            "required": "required",
            "type": "text",
            "value": "example-otp",
          }
        }
      />
    </mock-element>
  </mock-element>,
  <div
    className="auth-msg-wrapper"
  >
    <div
      style={
        {
          "backgroundColor": "#eeffee",
          "color": "green",
        }
      }
    >
      <span>
        OTP sent. Please check your email inbox.
      </span>
    </div>
  </div>,
  <div
    className="auth-btn-wrapper"
  >
    <div>
      <mock-element
        className="Button"
        data-props={
          {
            "className": "btn modal-btn auth send-otp",
            "disabled": false,
            "onClick": [Function],
          }
        }
      >
        Send OTP
      </mock-element>
      <mock-element
        className="Button"
        data-props={
          {
            "className": "btn modal-btn auth authorize",
            "disabled": false,
            "onClick": [Function],
          }
        }
      >
        Authorize
      </mock-element>
    </div>
  </div>,
]
`;
